config/jobs/kubernetes/cloud-provider-vsphere/cloud-provider-vsphere-config.yaml

```diff
@@ -384,6 +384,40 @@ presubmits:
       testgrid-dashboards: vmware-presubmits-cloud-provider-vsphere, presubmits-cloud-provider-vsphere-blocking
       testgrid-alert-email: k8s-testing-cloud-provider-vsphere+alerts@groups.vmware.com
 
+  # Executes the e2e tests with latest k8s version.
+  - name: pull-cloud-provider-vsphere-e2e-test-on-latest-k8s-version
+    decorate: true
+    labels:
+      preset-dind-enabled: "true"
+      # Borrows CAPV credentials for cloud provider e2e testing
+      preset-cloud-provider-vsphere-e2e-config-capv: "true"
+      preset-cloud-provider-vsphere-e2e-config: "true"
+      preset-kind-volume-mounts: "true"
+    branches:
+    - ^master$
+    path_alias: k8s.io/cloud-provider-vsphere
+    skip_submodules: true
+    always_run: false
+    run_if_changed: '\.go$|go.mod|hack/e2e\.sh|test/e2e/|charts/vsphere-cpi/'
+    skip_report: false
+    optional: true
+    spec:
+      containers:
+      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20240223-1ded72f317-master
+        command:
+        - runner.sh
+        args:
+        - ./hack/e2e.sh latest-k8s-version
```

Comment:
Is there anyway that we can check if this is master branch and we need to run `make latest-k8s-version`. Otherwise we will run 2 e2e tests, maybe there will be breaking changes between version N and N-1.

Reply:
I configured this pre-check job to run on PRs merging into master here: https://github.com/kubernetes/test-infra/pull/32105/files#diff-bcf1a86a8d03f4f33b85626569dfcccec6ffa492dd4189a6c842918606da4878R397, do you think it is good enough?